src/freenet/clients/http/ToadletContextImpl.java

```diff
@@ -376,15 +376,32 @@ static void sendReplyHeaders(OutputStream sockOutputStream, int replyCode, Strin
 			}
 		if(contentLength >= 0)
 			mvt.put("content-length", Long.toString(contentLength));
-		
-		String expiresTime;
+
+		boolean allowCaching; // For privacy reasons, only static
+							  // content may be cached
 		if (mTime == null) {
-			expiresTime = "Thu, 01 Jan 1970 00:00:00 GMT";
+			allowCaching = false;
+		} else {
+			allowCaching = true;
+		}
+		String expiresTime;
+		String cacheControl;
+		if (allowCaching) {
+			// use an expiry time of 30 day from now, about the frequency of Freenet releases
+			// Expires is needed for older browsers
+			expiresTime = TimeUtil.makeHTTPDate(System.currentTimeMillis() + DAYS.toMillis(30));
+			cacheControl = "public, max-age=" + String.valueOf(3600 * 24 * 30);
+			// Use contentlength as simplest possible hash function
+			// which needs no additional file access
+			mvt.put("ETag", String.valueOf(contentLength));
```

Comment:
I want to avoid messing too much with the structure of the code. We have neither the file content nor the file path at this point. This is not about cheaper, it is about avoiding to restructure code for this change.

I want to keep this pull-request short, and rather file a second one to improve on it.